### PR TITLE
JNG-4218 support add and remove for collections

### PR DESCRIPTION
**Pull request by Madmat8**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4218" title="JNG-4218" target="_blank">Sub-taskJNG-4218</a>  Add remove<Relation>, add<Relation> to embedded relations to DTO.
  </td></table>
  <br />

### CODE REVIEW

**Review by gaborflorian — APPROVED** [review]

**Review by robertcsakany — APPROVED** [review]

File: judo-runtime-core-jsl-itest/models/EntityRelationshipsModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/CompositionRelationshipsTest.java [hunk 1]
```diff
@@ -568,4 +557,139 @@ void testDeepCopyUpdate() {
...
+        assertEquals("C2", entityA1.getCollectionConA().stream().filter(c -> c.getStringC().orElseThrow().equals("C2")).findFirst().orElseThrow().getStringC().orElseThrow());
+        assertEquals("C3", entityA1.getCollectionConA().stream().filter(c -> c.getStringC().orElseThrow().equals("C3")).findFirst().orElseThrow().getStringC().orElseThrow());
+        assertEquals("C4", entityA1.getCollectionConA().stream().filter(c -> c.getStringC().orElseThrow().equals("C4")).findFirst().orElseThrow().getStringC().orElseThrow());
+        assertEquals("C5", entityA1.getCollectionConA().stream().filter(c -> c.getStringC().orElseThrow().equals("C5")).findFirst().orElseThrow().getStringC().orElseThrow());
+        assertEquals("C6", entityA1.getCollectionConA().stream().filter(c -> c.getStringC().orElseThrow().equals("C6")).findFirst().orElseThrow().getStringC().orElseThrow());
+
+        entityA1.addToCollectionConA(null);
+        assertEquals(6, entityA1.getCollectionConA().size());
+        EntityC c7 = EntityC.builder().withStringC("C7").build();
+        entityA1.addToCollectionConA(null, c7);
```
Comment (bencegelei) [inline judo-runtime-core-jsl-itest/models/EntityRelationshipsModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/CompositionRelationshipsTest.java:590]: I believe null values should not be permitted as they have the potential to obscure bugs in the event that variables unintentionally become null.

File: judo-runtime-core-jsl-itest/models/EntityRelationshipsModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/CompositionRelationshipsTest.java [hunk 2]
```diff
@@ -568,4 +557,139 @@ void testDeepCopyUpdate() {
...
+
+        EntityC c4 = EntityC.builder().withStringC("C4").build();
+        EntityC c5 = EntityC.builder().withStringC("C5").build();
+        EntityC c6 = EntityC.builder().withStringC("C6").build();
+        entityA1.addToCollectionConA(c4);
+        entityA1.addToCollectionConA(c5, c6);
+
+        assertEquals("C1", entityA1.getSingleRequiredConA().getStringC().orElseThrow());
+        assertEquals(5, entityA1.getCollectionConA().size());
+        assertEquals("C2", entityA1.getCollectionConA().stream().filter(c -> c.getStringC().orElseThrow().equals("C2")).findFirst().orElseThrow().getStringC().orElseThrow());
```
Comment (bencegelei) [inline judo-runtime-core-jsl-itest/models/EntityRelationshipsModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/CompositionRelationshipsTest.java]: I think the following would be easier to maintain and understand:

    assertTrue(entityA1.getCollectionConA().stream().anyMatch(c -> "C2".equals(c.getStringC().orElse(null))))

**Review by bencegelei — CHANGES_REQUESTED** [review]

File: judo-runtime-core-jsl-itest/models/EntityRelationshipsModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/CompositionRelationshipsTest.java (hunk 2 above)
Comment (Madmat8) [reply judo-runtime-core-jsl-itest/models/EntityRelationshipsModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/CompositionRelationshipsTest.java]: [12efd57](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/248/commits/12efd5755126ffb9485370c684ac86cd9acb336b)

File: judo-runtime-core-jsl-itest/models/EntityRelationshipsModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/CompositionRelationshipsTest.java (hunk 1 above)
Comment (Madmat8) [reply judo-runtime-core-jsl-itest/models/EntityRelationshipsModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/CompositionRelationshipsTest.java:590]: Currently, null values can also be specified in the set call. The reason I allowed it to add a null value in the add method was to make set and add consistent with each other.

**Review by noherczeg — APPROVED** [review]

**Review by bencegelei — APPROVED** [review]

**Pull request merged**